content-review/impl/turnitin-oc/src/main/java/org/sakaiproject/contentreview/turnitin/oc/ContentReviewServiceTurnitinOC.java

```diff
@@ -161,10 +166,132 @@ public void init() {
 		// Populate similarity report headers used in generateSimilarityReport
 		SIMILARITY_REPORT_HEADERS.putAll(BASE_HEADERS);
 		SIMILARITY_REPORT_HEADERS.put(HEADER_CONTENT, CONTENT_TYPE_JSON);
+		
+		// Populate webhook generation headers used in setupWebhook
+		WEBHOOK_SETUP_HEADERS.putAll(BASE_HEADERS);
+		WEBHOOK_SETUP_HEADERS.put(HEADER_CONTENT, CONTENT_TYPE_JSON);
 
 		// Populate content upload headers used in uploadExternalContent
 		CONTENT_UPLOAD_HEADERS.putAll(BASE_HEADERS);
 		CONTENT_UPLOAD_HEADERS.put(HEADER_CONTENT, CONTENT_TYPE_BINARY);
+
+		try {
+			// Get the webhook url
+			String webhookUrl = getWebhookUrl(Optional.empty());
+			// Hard-coded url for testing locally (leave in for now):
+			// String webhookUrl = "http://vericite.com/content-review-tool/webhooks?providerName=TurnitinOC";
+			// Get list of existing webhooks, if any
+			ArrayList<Webhook> webhooks = getWebhooks();
+
+			boolean webhooksSetup = false;
+			// Check to see if any webhooks exist for this url
+			for (Webhook webhook : webhooks) {
```

Comment:
you could combine this forloop and getwebhooks:

for (Webhook webhook : getWebhooks()) {